src/limited.rs

```diff
@@ -105,6 +105,16 @@ impl<T> Queue<T> {
     pub fn available(&self) -> isize {
         self.available.get()
     }
+    /// Await until the queue is full.
+    pub async fn full(&self) {
+        let capacity = self.capacity().try_into().unwrap();
+        let _ = self.pop_semaphore.acquire_many(capacity).await.unwrap();
```

Comment:
Tokio semaphores are fair in nature. This `acquire_many` call will actually block all other `acquire` calls until the queue is full. If you were to call this on a queue and you never put in `capacity` objects this call never returns and other calls to `pop` will be blocked indefinitely.